src/services/configServices/SettingsService.js

```diff
@@ -116,6 +154,37 @@ class SettingsService {
     }
   }
 
+  async updatePassword(sessionData, { encryptedPassword, iv, enablePassword }) {
+    const { siteName } = sessionData
+    const siteInfo = await this.sitesService.getBySiteName(siteName)
+    if (siteInfo.isErr()) {
+      return siteInfo
+    }
+    const { id, isPrivate } = siteInfo.value
+    if (isPrivate !== enablePassword) {
+      // For previously public repos, also need to set github repo to private
+      const privatiseRepoRes = await this.gitHubService.changeRepoPrivacy(
+        sessionData,
+        enablePassword
+      )
+      if (privatiseRepoRes.isErr()) return privatiseRepoRes
+      try {
+        await this.sitesService.update({
+          id,
+          isPrivate: enablePassword,
+        })
+      } catch (err) {
+        return errAsync(err)
+      }
+    }
```

Comment:
just wanted to note that we don't need the `try/catch` cos we still throw upstream :sadge: